src/adhocracy_core/adhocracy_core/evolution/__init__.py

```diff
@@ -898,8 +898,16 @@ def add_activation_config_sheet_to_user(root, registry):  # pragma: no cover
 @log_migration
 def add_global_anonymous_user(root, registry):  # pragma: no cover
     """Add  global anonymmous user."""
+    from pyramid.request import Request
     from adhocracy_core.resources.root import _add_anonymous_user
-    _add_anonymous_user(root, registry)
+    from adhocracy_core.interfaces import IRolesUserLocator
+    username = registry.settings.get('adhocracy.anonymous_user', 'anonymous')
+    request = Request.blank('/dummy')
+    request.registry = registry
+    adapter = registry.queryMultiAdapter((root, request), IRolesUserLocator)
+    anonymous_user = adapter.get_user_by_login(username)
+    if not anonymous_user:
+        _add_anonymous_user(root, registry)
 
```

Comment:
There is a helper script to get the anonymous user: adhocracy_core.resources.principals.get_anonymous_user